File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/per5python/archivo_farmacia.py
import pickle
from farmacia import Farmacia

class ArchivoFarmacia:
    def __init__(self, nombre_archivo):
        self.nombre_archivo = nombre_archivo

    def crear_archivo(self):
        """Crea el archivo si no existe"""
        try:
            with open(self.nombre_archivo, "wb") as archivo:
                pickle.dump([], archivo)
        except Exception as e:
            print("Error al crear archivo:", e)

    def guardar_farmacia(self, farmacia):
        """Guarda una farmacia en el archivo"""
        farmacias = self.leer_farmacias()
        farmacias.append(farmacia)
        with open(self.nombre_archivo, "wb") as archivo:
            pickle.dump(farmacias, archivo)

    def leer_farmacias(self):
        """Lee todas las farmacias guardadas en el archivo"""
        try:
            with open(self.nombre_archivo, "rb") as archivo:
                return pickle.load(archivo)
        except (FileNotFoundError, EOFError):
            return []

    def mostrar_medicamentos_tos(self, sucursal):
        """Mostrar los medicamentos para la tos de una sucursal específica"""
        farmacias = self.leer_farmacias()
        for farmacia in farmacias:
            if farmacia.sucursal == sucursal:
                print(f"Medicamentos para la tos en sucursal {sucursal}:")
                for med in farmacia.medicamentos:
                    if med.get_tipo().lower() == "tos":
                        print(med)

    def buscar_farmacia_medicamento(self, nombre_medicamento):
        """Mostrar sucursal y dirección de farmacias que tienen un medicamento específico"""
        farmacias = self.leer_farmacias()
        for farmacia in farmacias:
            if farmacia.busca_medicamento(nombre_medicamento):
                print(f"Sucursal: {farmacia.sucursal} | Dirección: {farmacia.direccion}")

File: repos_estudiantes_TARQUI_tercera_entrega/SIÑANI_QUELALI_IVAN/tarea3/per5java/ArchivoFarmacia.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class ArchivoFarmacia {
    private String nomArchivo;
    //Constructor
    public ArchivoFarmacia(String nomArchivo) {
        this.nomArchivo = nomArchivo;
    }
    //guardar una farmacia en archivo
    public void guardarFarmacia(Farmacia farmacia) {
        List<Farmacia> farmacias = leerFarmacias();
        farmacias.add(farmacia);
        try (ObjectOutputStream oos = new ObjectOutputStream(new FileOutputStream(nomArchivo))) {
            oos.writeObject(farmacias);
        } catch (IOException e) {
            System.out.println("Error al guardar farmacia: " + e.getMessage());
        }
    }
    //leer todas las farmacias guardadas en archivo
    public List<Farmacia> leerFarmacias() {
        try (ObjectInputStream ois = new ObjectInputStream(new FileInputStream(nomArchivo))) {
            return (List<Farmacia>) ois.readObject();
        } catch (EOFException | FileNotFoundException e) {
            return new ArrayList<>();
        } catch (IOException | ClassNotFoundException e) {
            System.out.println("Error al leer farmacias: " + e.getMessage());
            return new ArrayList<>();
        }
    }
    ///mostrar medicamentos para la tos de una sucursal específica
    public void mostrarMedicamentosTos(int sucursal) {
        List<Farmacia> farmacias = leerFarmacias();
        for (Farmacia farmacia : farmacias) {
            if (farmacia.getSucursal() == sucursal) {
                System.out.println("Medicamentos para la tos en sucursal " + sucursal + ":");
                for (Medicamento med : farmacia.medicamentos) {
                    if (med.getTipo().equalsIgnoreCase("Tos")) {
                        System.out.println(med);
                    }
                }
            }
        }
    }
    //buscar farmacias con un medicamento específico
    public void buscarFarmaciaMedicamento(String nombreMedicamento) {
        List<Farmacia> farmacias = leerFarmacias();
        for (Farmacia farmacia : farmacias) {
            if (farmacia.buscarMedicamento(nombreMedicamento) != null) {
                System.out.println("Sucursal: " + farmacia.getSucursal() + " | Dirección: " + farmacia.getDireccion());
            }
        }
    }
}
